### ISSUE TITLE
period in bucket name looks like it's messing up host

### ISSUE THREAD

**Issue by tjgillies** [danilop/yas3fs#137]
016-11-05 21:18:32,923 ERROR Uncaught Exception: <class 'ssl.CertificateError'> hostname 'tyler.indieweb.s3.amazonaws.com' doesn't match either of '*.s3.amazonaws.com', 's3.amazonaws.com' <traceback object at 0x10475fab8>

yas3fs s3://tyler.indieweb indieweb

**Comment by danilop** [comment]
Correct. The SSL certificate provided by AWS supports only one level of
subdomain under s3.amazonaws.com.

**Comment by dacut** [comment]
You should be able to fix this by changing the calling format you're using to call S3 -- `https://endpoint/bucket/object` instead of `https://bucket.endpoint/object`.

Looking at
[__init__.py](https://github.com/danilop/yas3fs/blob/master/yas3fs/__init__.py), there are two `boto.connet_s3()` calls around lines 151-153. Add the kw arg `calling_format=boto.s3.connection.OrdinaryCallingFormat`, and these issues should go away (without horrible hacks like 

Alas, the TLS spec says that *.foo only matches one level of domain names.

**Comment by dacut** [comment]
Turns out this was a tad more complex than I had anticipated, but I threw a pull request your way for this: https://github.com/danilop/yas3fs/pull/139

**Comment by Liath** [comment]
@tjgillies Could you pretty please test using latest?
```
git https://github.com/danilop/yas3fs.git
pip install ./yas3fs
```
pip might need sudo or to pass --user, idk your situation

**Comment by ldrolez** [comment]
Thanks it works!

**Comment by crudmucosa** [comment]
I think I am having the same issue: 

yas3fs picturesofus.net picturesofus
2017-07-20 19:17:06,450 INFO Version: 2.3.2
2017-07-20 19:17:06,450 INFO s3-retries: '3'
2017-07-20 19:17:06,450 INFO s3-retries-sleep: '1' seconds
2017-07-20 19:17:06,450 ERROR The S3 path to mount must be in URL format: s3://BUCKET/PATH, use -h for help.
cad-bane:picturesofusS3buckets fenris$ yas3fs s3://picturesofus.net picturesofus
2017-07-20 19:17:14,070 INFO Version: 2.3.2
2017-07-20 19:17:14,070 INFO s3-retries: '3'
2017-07-20 19:17:14,071 INFO s3-retries-sleep: '1' seconds
2017-07-20 19:17:14,071 INFO S3 bucket: 'picturesofus.net'
2017-07-20 19:17:14,071 INFO S3 prefix (can be empty): ''
2017-07-20 19:17:14,071 INFO Cache entries: '100000'
2017-07-20 19:17:14,071 INFO Cache memory size (in bytes): '134217728'
2017-07-20 19:17:14,071 INFO Cache disk size (in bytes): '1073741824'
2017-07-20 19:17:14,071 INFO Cache on disk if file size greater than (in bytes): '0'
2017-07-20 19:17:14,071 INFO Cache check interval (in seconds): '5'
2017-07-20 19:17:14,071 INFO Cache ENOENT rechecks S3: False
2017-07-20 19:17:14,071 INFO AWS Managed Encryption enabled: False
2017-07-20 19:17:14,071 INFO AWS Managed Encryption enabled: False
2017-07-20 19:17:14,071 INFO Number of parallel S3 threads (0 to disable writeback): '32'
2017-07-20 19:17:14,071 INFO Number of parallel downloading threads: '4'
2017-07-20 19:17:14,071 INFO Number download retry attempts: '60'
2017-07-20 19:17:14,071 INFO Download retry sleep time seconds: '1'
2017-07-20 19:17:14,071 INFO Number read retry attempts: '10'
2017-07-20 19:17:14,071 INFO Read retry sleep time seconds: '1'
2017-07-20 19:17:14,071 INFO Number of parallel prefetching threads: '2'
2017-07-20 19:17:14,071 INFO Download buffer size (in KB, 0 to disable buffering): '10485760'
2017-07-20 19:17:14,071 INFO Number of buffers to prefetch: '0'
2017-07-20 19:17:14,071 INFO Write metadata (file system attr/xattr) on S3: 'True'
2017-07-20 19:17:14,071 INFO Download prefetch: 'False'
2017-07-20 19:17:14,071 INFO Multipart size: '104857600'
2017-07-20 19:17:14,071 INFO Multipart maximum number of parallel threads: '4'
2017-07-20 19:17:14,071 INFO Multipart maximum number of retries per part: '3'
2017-07-20 19:17:14,071 INFO Default expiration for signed URLs via xattrs: '2592000'
2017-07-20 19:17:14,071 INFO S3 Request Payer: 'False'
2017-07-20 19:17:14,071 INFO Cache path (on disk): '/tmp/yas3fs/picturesofus.net'
2017-07-20 19:17:14,606 ERROR Uncaught Exception: <class 'ssl.CertificateError'> hostname 'picturesofus.net.s3.amazonaws.com' doesn't match either of '*.s3.amazonaws.com', 's3.amazonaws.com' <traceback object at 0x10455a950>
None

Tried the latest version and still get the above error.

Thanks for any help you can provide!

**Comment by crudmucosa** [comment]
Newest version works fine! Thanks much.
